### PR TITLE
Disable some tests from Windows IoT

### PR DESCRIPTION
**Pull request by danmosemsft**
Fix https://github.com/dotnet/corefx/issues/34004
Fix https://github.com/dotnet/corefx/issues/34000
Fix https://github.com/dotnet/corefx/issues/34030

Some trivial test disabling.

### CODE REVIEW

File: src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs [hunk 1]
```diff
@@ -1115,7 +1115,8 @@ public static TheoryData<bool> UseShellExecute
                 TheoryData<bool> data = new TheoryData<bool> { false };
 
                 if (   !PlatformDetection.IsUap // https://github.com/dotnet/corefx/issues/20204
-                    && !PlatformDetection.IsWindowsNanoServer) // By design
+                    && !PlatformDetection.IsWindowsNanoServer // By design
+                    && !PlatformDetection.IsWindowsIoTCore)
```
Comment (stephentoub) [inline src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs]: The previous line says "By design"... is this one by design, too?

File: src/System.Management/tests/WmiTestHelper.cs [hunk 2]
```diff
@@ -10,7 +10,11 @@ namespace System.Management.Tests
     public static class WmiTestHelper
     {
         private static readonly bool s_isElevated = AdminHelpers.IsProcessElevated();
-        private static readonly bool s_isWmiSupported = PlatformDetection.IsWindows && PlatformDetection.IsNotWindowsNanoServer && !PlatformDetection.IsUap;
+        private static readonly bool s_isWmiSupported =
+                                            PlatformDetection.IsWindows &&
+                                            PlatformDetection.IsNotWindowsNanoServer &&
+                                            PlatformDetection.IsNotWindowsIoTCore &&
+                                            !PlatformDetection.IsUap;
```
Comment (stephentoub) [inline src/System.Management/tests/WmiTestHelper.cs]: Curious why we have "IsNot" for some properties but then use "!" for others?

File: src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs (hunk 1 above)
Comment (danmosemsft) [reply src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs]: Yes

File: src/System.Management/tests/WmiTestHelper.cs (hunk 2 above)
Comment (danmosemsft) [reply src/System.Management/tests/WmiTestHelper.cs]: Existing inconsistency?

Comment (stephentoub) [reply src/System.Management/tests/WmiTestHelper.cs]: :)

**Comment by joperezr** [comment]
Thanks for fixing this Dan, just as FYI, I hear from the Engineering team that we are close to be able to run tests on arm32 as part of CI so we will be soon making a push to closing all of these issues.

**Pull request merged**